src/evm-block-extractor/src/main.rs

```diff
@@ -52,121 +43,56 @@ async fn main() -> anyhow::Result<()> {
     init_logger()?;
     let args = Args::parse();
 
-    let (start_block, append) = match get_last_block_number_from_output_file(&args.output_file) {
-        Some(last_block_number) => {
-            log::info!(
-                "last block number found in output file: {}",
-                last_block_number
-            );
-            (last_block_number + 1, true)
-        }
-        None => (args.start_block, false),
-    };
-    let end_block = args.end_block.unwrap_or(u64::MAX);
+    // Check if the dataset ID is valid
+    if args.dataset_id != "testnet" && args.dataset_id != "mainnet" {
+        return Err(anyhow::anyhow!(
+            "Invalid dataset ID. The dataset ID can be one of the following: testnet, mainnet"
+        ));
+    }
 
     log::info!("{PACKAGE}");
     log::info!("----------------------");
     log::info!("- rpc-url: {}", args.rpc_url);
-    log::info!("- output-file: {}", args.output_file.display());
-    log::info!("- start-block: {start_block:#x}");
-    log::info!("- end-block: {end_block:#x}");
+    log::info!("- dataset-id: {}", args.dataset_id);
+    log::info!("- max-number-of-requests: {}", args.max_number_of_requests);
     log::info!("----------------------");
 
     log::info!("initializing blocks-writer...");
-    let blocks_writer = BlocksWriter::new(&args.output_file, append)?;
+
     log::info!("blocks-writer initialized");
 
-    collect_blocks(
-        &args.rpc_url,
-        blocks_writer,
-        start_block,
-        end_block,
-        args.batch_size,
-    )
-    .await?;
+    let big_query_client = BigQueryBlockChain::new(args.dataset_id, args.sa_key).await?;
 
-    Ok(())
-}
+    let mut extractor = BlockExtractor::new(
+        args.rpc_url,
+        args.max_number_of_requests as u64,
+        args.rpc_batch_size,
+        Box::new(big_query_client.clone()),
+    );
 
-fn init_logger() -> anyhow::Result<()> {
-    env_logger::init();
+    let end_block = extractor.latest_block_number().await?;
+    log::debug!("latest block number: {}", end_block);
 
-    Ok(())
-}
+    let start_block = end_block.saturating_sub(MAX_EVM_BLOCKS);
```

Comment:
Can there be not loaded blocks older than `start_block`?

Reply:
Can you rephrase the question please ?